### PR TITLE
New RT convergence criteria

### PR DESCRIPTION
**Pull request by odespard**
Currently the graph of the proportion of precursors detected against RT error (which is used for automatic optimization) is relatively flat. This new set of convergence criteria and new way of finding the optimum (based on most narrow tolerance with proximity to the maximum). 

An example of the new behaviour is shown below, where the value taken forward for search is distinct from the value that maximises the number of precursors. 

image

NB: although this example shows the behaviour of the new rule, it is perhaps not the best illustration of why this change is useful. Often the curve is much flatter and the maximum is only slightly greater than all the other points, in which case choosing the narrower value is typically preferable. This change is also justified by the curve in our ground truth optimization (as shown in the precursor identifications in the plot below):
image

### CODE REVIEW

File: alphadia/constants/default.yaml
Comment (mschwoer) [inline alphadia/constants/default.yaml:257]: could you document here what these mean ?

File: alphadia/workflow/optimization.py
Comment (mschwoer) [inline alphadia/workflow/optimization.py:283]: alternative ordering:

```python
optimal_parameter = self.history_df["parameter"].loc[index_of_optimum]
self.workflow.optimization_manager.fit({self.parameter_name: optimal_parameter})

classifier_version_at_optimum = ...
self.workflow.optimization_manager.fit(
    {"classifier_version": classifier_version_at_optimum}
)
...

```
unless, of course, the `.fit()` operations interfere with the `*_at_optimum` values

File: alphadia/workflow/optimization.py
Comment (mschwoer) [inline alphadia/workflow/optimization.py:438]: (nit) from a method that start with `check_` I would always assume that something is raised if the check fails.. 
`_has_converged()`?

File: alphadia/workflow/optimization.py
Comment (mschwoer) [inline alphadia/workflow/optimization.py:457]: (nit) maybe `feature_history = self.history_df[self.feature_name]` improves readability even more?

File: alphadia/workflow/optimization.py
Comment (mschwoer) [inline alphadia/workflow/optimization.py:461]: if you go to `-3` here, should the check in l. 449 also be `<=3`?

**Review by mschwoer — APPROVED** [review]
LGTM!

File: alphadia/workflow/optimization.py
Comment (odespard) [reply alphadia/workflow/optimization.py:461]: No, because that will only stop returning False when we have at least three rows in the dataframe, at which point the -3 index should not pose any problems.

File: alphadia/workflow/optimization.py
Comment (odespard) [reply alphadia/workflow/optimization.py:438]: That's already the name of the attribute set by a conditional if self._check_convergence() is True. The easiest thing might be to make just_converged a property with the current definition of self._check_convergence(), and then we could dispense with the function. This is probably another PR though.

File: alphadia/workflow/optimization.py
Comment (odespard) [reply alphadia/workflow/optimization.py:461]: But I've now made it < 3 instead of <= 2 for clarity.

File: alphadia/workflow/optimization.py
Comment (odespard) [reply alphadia/workflow/optimization.py:474]: I'll add this in a chained PR where I also make this new convergence rule 1) configurable, and 2) available for all parameters.

File: alphadia/constants/default.yaml
Comment (odespard) [reply alphadia/constants/default.yaml:257]: I'll add this in a chained PR where I also make this new convergence rule 1) configurable, and 2) available for all parameters.

**Review by odespard — COMMENTED** [review]
I won't merge this until we merge the next chained PR (https://github.com/MannLabs/alphadia/pull/322), as I want to be able to use the old convergence criteria through the config.

File: alphadia/workflow/optimization.py
Comment (mschwoer) [inline alphadia/workflow/optimization.py:272]: (nit^3) 
turning the order of the first two (also in docstring) would make it even more readable :-)
```
        optimal_parameter = self.history_df["parameter"].loc[index_of_optimum]
        self.workflow.optimization_manager.fit({self.parameter_name: optimal_parameter})

        classifier_version_at_optimum = self.history_df["classifier_version"].loc[
            index_of_optimum
        ]
        self.workflow.optimization_manager.fit(
            {"classifier_version": classifier_version_at_optimum}
        )
```

**Review by mschwoer — APPROVED** [review]
LGTM!

**Pull request merged**
